include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -4900,6 +4900,10 @@ MGTwoLevelTransferNonNested<dim, VectorType>::prolongate_and_add_internal_comp(
     EvaluatorTypeTraits<dim, dim, n_components, Number>;
   using value_type = typename Traits::value_type;
 
+  std::array<Number, 8> typical_weights;
+  for (unsigned int i = 0; i < typical_weights.size(); ++i)
+    typical_weights[i] = Number(1) / Number(i + 1);
```

Comment:
Would you mean prefixing the loop with an assertion to check that the size is right?
```suggestion
  std::array<Number, 8> typical_weights;
  Assert (typical_weights.size() <= typical_weights.size(), ExcInternalError());
  for (unsigned int i = 0; i < typical_weights.size(); ++i)
    typical_weights[i] = Number(1) / Number(i + 1);
```

Reply:
I do not quite understand the assertion you plan for: I do not use this array for all accesses, just for the most likely ones, so the check for out-of-range access needs to happen further down. But I see the source of confusion, so I decided to add a comment at the first appearance, hoping that this would clarify the intent more clearly.

Reply:
Oh, I totally misread this. I *though* that what's happening is that use a fixed size 8 for `typical_weights`, but you index into it with `i` from zero to *some other array's size*. But you're using the same array, and so of course the assertion I mindlessly wrote down is entirely meaningless :-)

(I got my second dose of the shingles vaccine on Tuesday, and yesterday was out of it and close to falling asleep for most of the day -- not really a great experience, though it probably beats having shingles. In any case, this fits pretty well into what I perceived my cognitive abilities to be yesterday ;-) )

Reply:
I mean, at least I proposed to assert that `x<=x`. Imagine I had asked you to assert that `x<x`! What mayhem!